Add Frets On Fire

Unrud opened the pull request:
The upstream project seems unmaintained for 10 years.

Unrud commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
bot, build net.sourceforge.fretsonfire

barthalion commented:
Requires network to work:

```
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "/app/share/fretsonfire/src/MainMenu.py", line 103, in harness
    function(self, *args, **kwargs)
  File "/app/share/fretsonfire/src/MainMenu.py", line 135, in newSinglePlayerGame
    self.engine.startServer()
  File "/app/share/fretsonfire/src/GameEngine.py", line 271, in startServer
    self.server = Server(self)
  File "/app/share/fretsonfire/src/Server.py", line 33, in __init__
    Network.Server.__init__(self)
  File "/app/share/fretsonfire/src/Network.py", line 179, in __init__
    self.bind((localOnly and "localhost" or "", port))
  File "/app/lib/python2.7/asyncore.py", line 342, in bind
    return self.socket.bind(addr)
  File "/app/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
gaierror: [Errno -3] Temporary failure in name resolution
```

Unrud commented:
> Requires network to work:

It only uses the loopback network interface. Can you try running:

```
flatpak run --command=python2 net.sourceforge.fretsonfire -c "import socket; socket.getaddrinfo('localhost', 0)"
```

barthalion commented:
Fails as well. Maybe some nsswitch misconfiguration on my side.

LeandroStanger commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
bot, build net.sourceforge.fretsonfire

LeandroStanger commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
This should be in good shape now.
I created new patches for several bugs, that cause crashes and removed most of the patches from Debian (they were either unnecessary or changed gameplay).
I kept the patches from Debian that revert font rendering to an older version (1.2.451) and make it run faster. The new version of font rendering is glitchy on Intel and can even crash the X server.
I also patched out the highscore upload (the servers are down) and replaced ``localhost`` with an IP (@barthalion).

nedrichards commented on net.sourceforge.fretsonfire.appdata.xml:
```diff
@@ -0,0 +1,96 @@
+    <ul>
+      <li>128 MB of RAM</li>
+      <li>A fairly fast OpenGL graphics card with decent drivers</li>
+      <li>Windows: Direct X compatible sound card</li>
```
Could you remove the non-Linux requirements from the appdata? If we're patching it in then we don't have to be cross platform

Unrud replied:
done

LeandroStanger commented:
bot, build net.sourceforge.fretsonfire

Unrud commented:
bot, build net.sourceforge.fretsonfire

barthalion commented:
bot, build net.sourceforge.fretsonfire

nedrichards commented:
bot, build net.sourceforge.fretsonfire

nedrichards commented:
Repository has been created: https://github.com/flathub/net.sourceforge.fretsonfire

Linkid commented:
Hi. If you want, FoFiX is a fork of Frets On Fire and is its logical successor since FoF is not maintained anymore : https://github.com/fofix/fofix ;).

Unrud commented:
@Linkid You can publish FoFiX on Flathub. (I think you have to release stable version first.) It would be nice to have both the classic FoF and FoFix available.
